src/shared/services/form/field-type-sanitiser.ts

```diff
@@ -13,33 +13,33 @@ export class FieldTypeSanitiser {
    */
    sanitiseLists(caseFields: CaseField[], editForm: any) {
 
-    this.getDynamicListsFromCaseFields(caseFields).forEach(dynamicField => {
-      this.getListOfKeysFromEditForm(editForm).forEach((key) => {
-        this.createValueCodePairAlongWithListIfKeyExistsInForm(dynamicField, key, editForm);
-      });
+    caseFields.forEach(caseField => {
+      if (caseField.field_type.type === 'DynamicList') {
+        this.getListOfKeysFromEditForm(editForm).forEach((key) => {
+          this.createValueCodePairAlongWithListIfKeyExistsInForm(caseField, key, editForm);
+        });
+      } else if (caseField.field_type.type === 'Complex') {
```

Comment:
please use SERVER_RESPONSE_FIELD_TYPE_COMPLEX instead of literals

Reply:
done